types/ink-text-input/index.d.ts

```diff
@@ -0,0 +1,17 @@
+// Type definitions for ink-text-input 2.0
+// Project: https://github.com/vadimdemedes/ink-text-input#readme
+// Definitions by: Łukasz Ostrowski <https://github.com/lukostry>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.8
+
+import { Component } from 'ink';
+
+export interface TextInputProps {
+    focus?: boolean;
+    onChange?: (value: string) => void;
+    onSubmit?: (value: string) => void;
+    placeholder?: string;
+    value?: string;
+}
+
+export default class TextInput extends Component<TextInputProps> { }
```

Comment:
`export = class TextInput`... is a more appropriate export for this given [the source](https://github.com/vadimdemedes/ink-text-input/blob/57d3d387add3670e9be14772ffb927a3ada87fbc/src/index.js#L77)

the corresponding import would be `import TextInput = require('TextInput')`

Reply:
`module.exports = TextInput;` implies it's not a named export. Compare usage provided in the original repo: https://github.com/vadimdemedes/ink-text-input/:

```
const {h, render, Component} = require('ink');
const TextInput = require('ink-text-input');
```
TextInput is imported as a default value from `ink-text-input` package.

Or even `react`, it's exported using example the same syntax: https://github.com/facebook/react/blob/master/packages/react/index.js#L16 and in `babel` ecosystem it's always imported as a default value, not a named: `import React from 'react'`. 

All in all, if you can change that in type definition, but the code compiled to `commonjs` by `tsc` won't work as expected.

Reply:
Agreed it's not a named export, it's an export assignment so `export = ` is still the right way to go. It's a CommonJS module.

See `react`'s [type definitions](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L46) for reference.

The `import react from 'react'` syntax is common, but only usable when your bundler supports CommonJS - ES module interop. In that case,  you'd want to set the `--allowSyntheticDefaultImports` compiler flag when you use the typings and everything would be fine. But since that's environment dependent, it's not something that should be encoded in the typings on DT. You want the typings to reflect the "truth", i.e. that this is a CommonJS module.

Reply:
I agree, it will work with `allowSyntheticDefaultImports` set to `true`. I could provide requested change, but I'm worried most people won't know this needs not be changed (since it's not a default compiler option) and I will assume those typings are incorect (since original repo is written in JS and there no example how to use it with TS).

WDYT?

Reply:
Yeah, I share your concern that it's not obvious to users, and most people are already pretty confused about modules. That's certainly something we've been working to do better with TS.

However I think consistency and accuracy wins here. There is lots of precedent for this exact issue so it's better for these typings to stay consistent with the rest of them on DT.

In short, I would still really like the `export =` change :)

Reply:
OK, will change then.

Reply:
I've updated the file definition, I've tried to strictly follow README and used definitions for `chai-http` as a reference:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/ffbf9da05fd9fe404a250586b15b7a3276bfe7cc/types/chai-http/index.d.ts#L77-L78

Please let me know if that's alright now, if yes, I would update my other PR as well: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/33029

Reply:
Looks great thanks!